src/pybind/mgr/dashboard_v2/models/nodb.py

```diff
@@ -0,0 +1,897 @@
+        self.primary_key = primary_key
+        self.editable = editable
+        self.null = null
+        self.blank = blank
+        self.default = default
+        self.help_text = help_text
+        self.validators = validators
+        self.verbose_name = verbose_name
+        self.choices = choices
+        self.model = None
+        self.name = name
+        self.attname = None
+
+    def set_attributes_from_name(self, name):
+        if not self.name:
+            self.name = name
+        self.attname = name
+        if self.verbose_name is None and self.name:
+            self.verbose_name = self.name.replace('_', ' ')
+
+    def contribute_to_class(self, cls, name):
+        self.set_attributes_from_name(name)
+        self.model = cls
+        cls._meta.add_field(self)
+        if self.choices:
+            setattr(cls, 'get_%s_display' % self.name,
+                    partial(cls._get_FIELD_display, field=self))
+
+    def _get_FIELD_display(self, field):
+        value = getattr(self, field.attname)
+        return str(dict(field.flatchoices).get(value, value))
+
+    def get_default(self):
+        return self.default if self.has_default() else None
+
+    def has_default(self):
+        return self.default is not NOT_PROVIDED
+
+
+class JsonField(Field):
+    empty_strings_allowed = False
+
+    def __init__(self, *args, **kwargs):
+        """
+        :param base_type: list | dict
+        :type base_type: type
+        :rtype: JsonField[T]
+        """
+        self.base_type = kwargs['base_type']
+        del kwargs['base_type']
+        super(JsonField, self).__init__(*args, **kwargs)
+
+    def to_python(self, value):
+        """:rtype: T"""
+        def check_base_type(val):
+            if not isinstance(val, self.base_type):
+                raise ValidationError(
+                    "invalid JSON type. Got {}, expected {}".format(type(parsed), self.base_type))
+            return val
+
+        if value is None:
+            return None
+        if isinstance(value, self.base_type):
+            return value
+        if not value and self.null:
+            return None
+        try:
+            parsed = json.loads(value)
+            return check_base_type(parsed)
+        except (ValueError, TypeError) as _:
+            raise ValidationError("invalid JSON")
+
+    @property
+    def empty_values(self):
+        return ['', [], {}]
+
+
+class IntegerField(JsonField):
+    def __init__(self, **kwargs):
+        super(IntegerField, self).__init__(base_type=int, **kwargs)
+
+
+class CharField(JsonField):
+    def __init__(self, choices=None, **kwargs):
+        super(CharField, self).__init__(base_type=str, **kwargs)
+
+
+class BooleanField(JsonField):
+    def __init__(self, **kwargs):
+        super(BooleanField, self).__init__(base_type=bool, **kwargs)
+
+
+class FloatField(JsonField):
+    def __init__(self, **kwargs):
+        super(FloatField, self).__init__(base_type=float, **kwargs)
+
```

Comment:
Remove one of the 3 blank lines.

Reply:
I took care of all linting errors today.